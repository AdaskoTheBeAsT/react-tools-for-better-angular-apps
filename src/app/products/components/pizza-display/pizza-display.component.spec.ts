import { async, ComponentFixture, TestBed } from '@angular/core/testing';
// import { AnimationTriggerMetadata } from '@angular/animations';

import { PizzaDisplayComponent } from './pizza-display.component';
import { Pizza, Topping } from '../../models';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

describe(`${PizzaDisplayComponent.name}`, () => {
  let component: PizzaDisplayComponent;
  let fixture: ComponentFixture<PizzaDisplayComponent>;
  const toppingsMock: Topping[] = [
    {
      id: 1,
      name: 'anchovy',
    },
    {
      id: 2,
      name: 'bacon',
    },
  ];
  const pizzaMock: Pizza = {
    id: 1,
    name: 'Pepperoni pizza',
    toppings: [],
  };

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        declarations: [PizzaDisplayComponent],
        imports: [NoopAnimationsModule],
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(PizzaDisplayComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it(`should display pizza`, () => {
    component.pizza = pizzaMock;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it(`should display pizza with toppings`, () => {
    const pizza = {
      ...pizzaMock,
      toppings: toppingsMock,
    };
    component.pizza = pizza;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
    expect(component.pizza.toppings).toEqual(toppingsMock);
  });
});
