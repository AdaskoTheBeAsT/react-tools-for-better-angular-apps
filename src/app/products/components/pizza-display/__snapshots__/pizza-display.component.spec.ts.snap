// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaDisplayComponent should display pizza 1`] = `
<pizza-display
  pizza={[Function Object]}
>
  
      <div
    class="pizza-display"
  >
    
        
    <div
      class="pizza-display__base"
    >
      
          
      <img
        class="ng-tns-c1-0"
        src="/assets/img/pizza.svg"
      />
      
          
      
      
        
    </div>
    
      
  </div>
    
</pizza-display>
`;

exports[`PizzaDisplayComponent should display pizza with toppings 1`] = `
<pizza-display
  pizza={[Function Object]}
>
  
      <div
    class="pizza-display"
  >
    
        
    <div
      class="pizza-display__base"
    >
      
          
      <img
        class="ng-tns-c2-0"
        src="/assets/img/pizza.svg"
      />
      
          
      
      
        
    </div>
    
      
  </div>
    
</pizza-display>
`;
